File: buildSrc/src/main/java/com/chenxuan/gradle/CustomPlugin.java
package com.chenxuan.gradle;

import com.android.build.gradle.AppPlugin;
import com.android.build.gradle.BaseExtension;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

public class CustomPlugin implements Plugin<Project> {

    @Override
    public void apply(Project project) {
        boolean isApp = project.getPlugins().hasPlugin(AppPlugin.class);
        if (isApp) {
            project.getExtensions().findByType(BaseExtension.class)
                    .registerTransform(new CustomTransform());
        }
    }
}

File: buildSrc/src/main/java/com/chenxuan/gradle/CustomPlugin.kt
package com.chenxuan.gradle

import com.android.build.gradle.AppPlugin
import com.android.build.gradle.BaseExtension
import com.chenxuan.gradle.image.ImageTransform
import org.gradle.api.Plugin
import org.gradle.api.Project

class CustomPlugin : Plugin<Project> {
    override fun apply(project: Project) {
        val isApp = project.plugins.hasPlugin(AppPlugin::class.java)
        if (isApp) {
            project.extensions.findByType(BaseExtension::class.java)
                ?.registerTransform(ImageTransform())
        }
    }
}